### ISSUE TITLE
Some output goes missing when stdout isn't a terminal

### ISSUE THREAD

**Issue by markshep** [ios-control/ios-deploy#87]
The output of ios-deploy differs depending on whether its stdout is a terminal or not.  When its stdout is a TTY all looks good with the install messages going right up to 100% and there being a "Debug phase" section.  However when stdout is not a TTY the install progress messages stop a little bit short of the end of copying the files, there's no "Debug phase" output and the output starts again from the "Executing commands in ..." line.

It seems to be that when lldb is started any text that was pending in the stdout buffer is discarded rather than being displayed.  So I fixed this by adding the following line near the start of main(), just before the getopt_long() call:

```
    setbuf(stdout, NULL);
```

That has the added advantage that all printf() calls are immediately outputted so that any calling program that's reading ios-deploy's output can see the very latest output from without having to wait for the stdout buffer to fill up and an auto-flush to occur.  Please could this change be made to master too?

Here are my system details:

```
% sw_vers -productVersion
10.10.1
% ios-deploy -V
1.3.2
% xcodebuild -version
Xcode 6.1
Build version 6A1052d
% xcode-select --print-path
/Applications/Xcode.app/Contents/Developer
% gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.0.0
Thread model: posix
% lldb --version
lldb-320.4.152
```

**Comment by ryanwilliams83** [comment]
I've encountered a problem which is possibly related.
Running a cordova app using ios-deploy in a SSH (putty) session.
Some calls to console.log() in JS result in only the end of the string being printed in the lldb window.
Problem only seems to occur with large strings.
I've tried --unbuffered but it didn't resolve the issue for me.
Anyone with any idea or suggestion to where/who I can ask for assistance can contact me at
ryan (at) busivid.com
